tools/logger/logger.c

```diff
@@ -140,9 +142,39 @@ static int configure_uart(const char *file, unsigned int baud)
 	return ret < 0 ? -errno : fd;
 }
 
+/* Concantenate `config->filter_config` with `input` + `\n` */
+static int append_filter_config(struct convert_config *config,
+				const char *input)
+{
+	char *buff;
+	size_t len;
+
+	if (!config->filter_config) {
```

Comment:
Can you add a comment each block here, I'm trying to understand why we cant use sprintf() here.